modules/tinymce/src/plugins/directionality/main/ts/core/Direction.ts

```diff
@@ -5,24 +5,48 @@
  * For commercial licenses see https://www.tiny.cloud/
  */
 
+import { Arr, Optional } from '@ephox/katamari';
+import { Traverse, Attribute, SugarElement, SugarNode, SelectorFind, Direction, SelectorFilter } from '@ephox/sugar';
+
 import Editor from 'tinymce/core/api/Editor';
-import Tools from 'tinymce/core/api/util/Tools';
 
-const setDir = (editor: Editor, dir: string) => {
-  const dom = editor.dom;
-  let curDir;
-  const blocks = editor.selection.getSelectedBlocks();
+type Dir = 'rtl' | 'ltr';
 
-  if (blocks.length) {
-    curDir = dom.getAttrib(blocks[0], 'dir');
+const getParentElement = (element: SugarElement<Element>): Optional<SugarElement<Element>> => {
+  return Traverse.parent(element).filter(SugarNode.isElement);
+};
```

Comment:
```suggestion
const getParentElement = (element: SugarElement<Element>): Optional<SugarElement<Element>> =>
  Traverse.parent(element).filter(SugarNode.isElement);
```
We normally try to use shorthand if possible, unless it causes issues with readability.

Reply:
Changed.

Reply:
Not fixed sorry, as you squashed this all onto a single line which is too long and not what I had suggested.

Reply:
Changed to two lines. What's the line width?